rs/execution_environment/src/execution_environment/tests.rs

```diff
@@ -804,6 +804,44 @@ fn get_canister_status_of_nonexisting_canister() {
     assert_eq!(ErrorCode::CanisterNotFound, err.code());
 }
 
+#[test]
+fn get_canister_status_memory_metrics() {
```

Comment:
This is an interesting way you found to check the changes. I'm wondering though if this is future-proof. We kinda add a dependency that (1) there is a total memory_size field we can check against (but we might want to retire in the future) and (2) the breakdown is summing up to the total (true now but if in the future we decided to not include something in the breakdown, it would also break the test).

Did you try the approach we discussed where you would trigger some memory usage increase and then try to check whether that's observable in canister status as expected?